src/inverted-index.js

```diff
@@ -1,42 +1,138 @@
+  /**
+   * Checks for malformed JSON file
+   * @param {object} jsonFile - the JSON file to be checked
+   * @returns {boolean} - either true or false
+   */
+  static isMalformed(jsonFile) {
     let result = false;
     for (let i = 0; i < jsonFile.length; i += 1) {
       result = !('title' in jsonFile[i] && 'text' in jsonFile[i]);
-      if (result === true) break;
+      if (result) break;
     }
     return result;
   }
 
-  tokenize(text) {
+  /**
+   * Breaks a string of text into unique word tokens
+   * @param {string} text - the string to be broken
+   * @returns {set} - a set of unique tokens
+   */
+  static tokenize(text) {
     text = new Set(text.toLowerCase().split(' '));
     return Array.from(text);
   }
 
+  /**
+   * Gets all the content of JSON file
+   * @param {array} jsonFile - array of book objects
+   * @returns {string} - a string mashup of both title and text
+   */
+  static flattenContent(jsonFile) {
+    let flattenedContent = '';
+    jsonFile.forEach((book) => {
+      flattenedContent += `${book.title} ${book.text} `;
+    });
+    return flattenedContent.trim().toLowerCase();
+  }
+
+  /**
+   * Reads the file and creates an index of the words in it
+   * @param {string} fileName - the name of the book to be indexed
+   * @param {string} fileContent - the content of the file (JSON array)
+   * @returns {string} - returns a string
+   */
   createIndex(fileName, fileContent) {
-    if (!this.isValid()) {
+    if (!InvertedIndex.isValid(fileContent)) {
       return 'The uploaded file is invalid';
     }
-
-    if (this.isMalformed()) {
+    if (InvertedIndex.isMalformed(fileContent)) {
       return 'The JSON file is malformed';
     }
 
-    const book1Text = this.tokenize(fileContent[0].text);
-    const book2Text = this.tokenize(fileContent[1].text);
+    const index = {};
+    const allContent = InvertedIndex
+    .tokenize(InvertedIndex.flattenContent(fileContent));
+    let eachContent;
+
+    fileContent.forEach((book, i) => {
+      eachContent = book;
+      eachContent = new Set(InvertedIndex
+      .tokenize(`${eachContent.title} ${eachContent.text}`));
+
+      allContent.forEach((word) => {
+        if (eachContent.has(word)) {
+          if (word in index) index[word].push(i);
+          else index[word] = [i];
+        }
+      });
+    });
+    this.indices[fileName] = index;
+    return JSON.stringify(index);
   }
 
-  searchIndex() {
-    
+  /**
+   * Searches the already created index
+   * @param {string} fileName - name of the file to be searched
+   * @param {string|array} terms - search terms
+   * @returns {object} - contains the location of each terms
+   */
+  searchIndex(fileName = 'all', ...terms) {
+    const indices = this.indices;
+    const searchTerms = [];
+    const result = {};
+
+    terms.forEach((term) => {
+      if (Array.isArray(term)) searchTerms.push(...term);
+      if (typeof term === 'string') searchTerms.push(...term.split(' '));
+    });
+
+    /**
```

Comment:
It is not appropriate to have multiline comment in your methods. Your jsdoc should cover that. I got this feedback from my LFA